dev-lang/go/go-9999.ebuild

```diff
@@ -123,17 +127,30 @@ src_compile()
 	export GOHOSTARCH=$(go_arch ${CBUILD})
 	export GOHOSTOS=$(go_os ${CBUILD})
 	export CC=$(tc-getBUILD_CC)
+	export GOOS=$(go_os)
+
+	cd src
+
+	# Build the target support first.
```

Comment:
As noted in a comment below this one, the host build needs to be done last so that the go environment is correct for native builds.  If we build for the host then build for the target the go environment will be for the last target.